hoot-core/src/main/cpp/hoot/core/io/OsmApiChangeset.cpp

```diff
@@ -160,9 +160,22 @@ void XmlChangeset::loadElements(QXmlStreamReader& reader, ChangesetType changese
       {
         long id = reader.attributes().value("ref").toString().toLong();
         std::dynamic_pointer_cast<XmlWay>(element)->addNode(id);
+        //  Update the node to way map
+        _nodeIdsToWays[id].insert(element->id());
```

Comment:
The changeset now keeps track of each node's list of parent ways and relations.  Same with ways and relations below.